artifact/v3/scripts/executor.hpp

```diff
@@ -107,19 +109,34 @@ class ScriptRunner {
 		HandlerFunction handler);
 
 	void HandleScriptError(Error err, HandlerFunction handler);
+	void HandleScriptRetry(
+		vector<string>::iterator current_script,
+		vector<string>::iterator end,
+		bool ignore_error,
+		HandlerFunction handler);
+	void HandleScriptNext(
+		vector<string>::iterator current_script,
+		vector<string>::iterator end,
+		bool ignore_error,
+		HandlerFunction handler);
+	void MaybeSetupRetryTimeoutTimer();
 
 	string ScriptPath(State state);
 
 	events::EventLoop &loop_;
 	bool is_artifact_script_;
-	chrono::seconds state_script_timeout_;
+	chrono::milliseconds script_timeout_;
+	chrono::milliseconds retry_interval_;
+	chrono::milliseconds retry_timeout_;
 	string artifact_script_path_;
 	string rootfs_script_path_;
 	processes::OutputCallback stdout_callback_;
 	processes::OutputCallback stderr_callback_;
 	Error error_script_error_;
 	vector<string> collected_scripts_;
 	unique_ptr<processes::Process> script_;
+	unique_ptr<events::Timer> retry_interval_timer_;
+	unique_ptr<events::Timer> retry_timeout_timer_;
```

Comment:
My gut feeling tells me that doing allocation and deallocation to avoid a `bool` flag on the timer is a waste of resources. Not a biggie, of course, but I think adding that `bool active_` flag and `bool Timer::GetActive() { return active_ }` wouldn't be that much work.